### PR TITLE
Check if :file is really a file

### PR DESCRIPTION
**Pull request by yyoncho**
- fixes https://github.com/emacs-lsp/lsp-ui/issues/217

#

### CODE REVIEW

**Comment by MatthewZMD** [comment]
A general question. Is there any guidelines or rule of thumb for lsp team on PR code reviews and testing?

**Comment by yyoncho** [comment]
Do you have something particular in mind?

**Comment by MatthewZMD** [comment]
I'm wondering what's the standard way to test the `needs testing` tagged PRs in the lsp-ui context.

**Comment by brotzeit** [comment]
I just added this label to signal that I don't just ignore a pull request ;)

The idea was that somebody needs to test it manually. Automatic tests are always nice, but I think most stuff needs to be tested manually in case of lsp-ui. But maybe we find at least some solutions for writing tests. No tests at all sucks...

**Comment by MatthewZMD** [comment]
Gotcha

File: lsp-ui-peek.el [hunk 1]
```diff
@@ -698,12 +698,18 @@ references.  The function returns a list of `ls-xref-item'."
...
-    (mapcar #'lsp-ui-peek--get-xrefs-list
-            (if (gethash "uri" (car locs))
-                ;; Location[]
-                (--group-by (lsp--uri-to-path (gethash "uri" it)) locs)
-              ;; LocationLink[]
-              (--group-by (lsp--uri-to-path (gethash "targetUri" it)) locs)))))
+    (-filter
+     (-lambda ((&plist :file))
+       (or (f-file? file)
+           (prog1 nil
```
Comment (MaskRay) [inline lsp-ui-peek.el]: `(prog1 nil` -> `(ignore`

Comment (yyoncho) [reply lsp-ui-peek.el]: I wonder which one is the recommended solution here: `ignore` is shorter/more in a functional style and slower while prog1 is more imperative/faster and longer. I am interested to hear what elisp experts recommend.

Comment (brotzeit) [reply lsp-ui-peek.el]: Why is prog1 faster ? No elisp expert at all :P

Comment (yyoncho) [reply lsp-ui-peek.el]: prog1 is primitive written in C and there is no function invocation.

**Comment by yyoncho** [comment]
Addressed the comment.

**Pull request merged**